### PR TITLE
Pin spacy transformers version

### LINKED ISSUE #75: Spacy-transformers incompatible with transformers
# Problem
At the moment when installing WellcomeML, the latest version of spacy transformers is installed which creates a conflict with the transformers requirement of below 2.
```
ERROR: spacy-transformers 0.6.1 has requirement spacy<2.4.0,>=2.3.0, but you'll have spacy 2.2.1 which is incompatible.
ERROR: spacy-transformers 0.6.1 has requirement transformers<2.9.0,>=2.4.0, but you'll have transformers 2.0.0 which is incompatible.
ERROR: tensorflow 2.2.0 has requirement scipy==1.4.1; python_version >= "3", but you'll have scipy 1.5.0 which is incompatible.
```

# Replicating
You can replicate by installing the latest version of WellcomeML with deep learning. Downloading the models (which for some reason it does not work automatically, possibly another issue) and try to import wellcomeml.ml
```
python3 -m venv venv
source venv/bin/activate
pip install wellcomeml[deep-learning]
python -m wellcomeml download models
python -c "import wellcomeml.ml"
```

this throws
```
pkg_resources.VersionConflict: (transformers 2.0.0 (/Users/nicksorros/code/test/venv/lib/python3.7/site-packages), Requirement.parse('transformers<2.9.0,>=2.4.0'))
```

Interestingly the tests do not fail since when creating the virtualenv we do not install the deep learning requirements from the setup.py but the requirements.txt.

# Solution
A temporary fix is to pin the spacy transformers library to 0.5.1. An alternative is to ensure the library works with the latest transformers release.

Additionally when creating the virtualenv we should install the deep learning requirements from the setup.py `pip install -e .[deep-learning]`. 

Lastly, maintaining a requirements.txt and a setup.py creates an inconsistency since when a user installs the dependencies are defined by setup.py but when we develop or run tests we use mostly the requirements (as this runs first anyway). So removing requirements.txt all together will resolve future bugs similar to this. An alternative would be to read the requirements from setup.py but separating them to extras vs no extras could be a bit painful

### PR DESCRIPTION
**Pull request by nsorros**
Fixes #75 where the problem is described in more detail.

TL;DR. This PR removes requirements.txt and pins the version of spacy transformers. The first ensures that similar failures can be cached from tests and the second resolves a version conflict that the latest release of spacy-transformers introduced

### CODE REVIEW

**Comment by aCampello** [comment]
I am happy with only having setup.py, but I we might need to include other packages explicitly to make it work, for example `Cython` (see pipeline logs)

File: Makefile [hunk 1]
```diff
@@ -7,30 +7,13 @@ $(VIRTUALENV)/.installed:
 	@if [ -d $(VIRTUALENV) ]; then rm -rf $(VIRTUALENV); fi
 	@mkdir -p $(VIRTUALENV)
 	virtualenv --python $(PYTHON_VERSION) $(VIRTUALENV)
-	$(VIRTUALENV)/bin/pip3 install -r requirements.txt
 	$(VIRTUALENV)/bin/pip3 install -r requirements_test.txt
-	$(VIRTUALENV)/bin/pip3 install -e .
-	$(VIRTUALENV)/bin/pip3 install git+https://github.com/epfml/sent2vec.git # otherwise it fails cause setup.py relies on cython and numpy being installed
```
Comment (nsorros) [inline Makefile:13]: I think the problem @aCampello was caused by removing this line not requirements. Testing that theory now.

Comment (aCampello) [reply Makefile:13]: I think the problem is that when we download sent2vec it doesn't install the dependencies of that project.

Comment (nsorros) [reply Makefile:13]: Issue #83 and resolved by #84

Comment (aCampello) [reply Makefile:13]: 👍

**Comment by aCampello** [comment]
Does this also solve  https://github.com/wellcometrust/WellcomeML/issues/33?

**Comment by nsorros** [comment]
> Does this also solve #33?

How so?

**Comment by aCampello** [comment]
Nevermind, I thought the new version of transformers somehow installed things on the fly as well. But I think we can resolve that issue quite soon anyway.
